### ISSUE TITLE
Never finish

### ISSUE THREAD

**Issue by emadmahdi** [GhostofGoes/getmac#70]
Hi,
I installed your "getmac" module v 0.8.2 on Android 6.0.1 using python 3.8 installed on an ASUS Phone with wifi turned on and data also truned on ... But I got this problem:

When i execute the module ... it never finish and never stop ... After debugging I found that the below lines where the casue of this problem ... Hence I commited them and the module worked again normally ... I've seen these kind of probelms resulting from regex expression problems.

I also noticed that your module does not produce any mac address for this android phone ... It only give "NoneType" ?!!!

If you need to test a fix ... let me know i can help.

```
#### ip link (Don't use 'list' due to SELinux [Android 24+])
####(
####    to_find + r".*\n.*link/ether " + MAC_RE_COLON,
####    0,
####    "ip",
####    ["link %s" % to_find, "link"],
####),
```

**Comment by emadmahdi** [comment]
To fix this issue and similar issues on other systems ... I suggest that you run all your methods at the same time using threading ... hence if one thread succeeded then simply stop all threads and return the found mac address ... but if threads took more than 5 seconds without any results then simply kill the threads and return failed finding mac address.

**Comment by emadmahdi** [comment]
something simple like this is enoguh:

```
results = ''
def my_getmac():
	global results
	import getmac
	results = getmac.get_mac_address()
	return

def check():
	import threading
	th1 = threading.Thread(target=my_getmac,args=())
	th1.start()
	for ii in range(5*4):
		if not results: time.sleep(0.25)
		else:
			mac = results
			break
	else: mac = ''
```

**Comment by GhostofGoes** [comment]
I've never actually tested it on Android, so while it's unfortunate it's not working I'm also not surprised. The code mentioning Android was based on code from CPython's `uuid` module. 

To prevent it returning None you need to find a command that will get a MAC on your version of Android, then add a parser for it.

Regarding threading...I'm currently in the process of refactoring the codebase to be smarter about how methods are selected and executed, which has drastically improved performance (in the limited testing I've done so far), which will make threading (hopefully) unnecessary. The `refactor` branch has the changes. Don't have an exact date when the refactor will be merged in but it's getting close, just need to finish implementing a few things, finish updating tests, and cut a beta.

**Comment by emadmahdi** [comment]
I think you still need threading for timeout unless you have another method to implement the timeout

**Comment by emadmahdi** [comment]
I checked my Android ... and this command is the cause for hangging:
ip link SOMETHING

These two commands is working perfectly on my Android phone:
ip link
ip addr

So, I replaced your old lines with this line and your getmac module started to work:
(r"ether " + MAC_RE_COLON, 0, "ip", ["link","addr"]),

**Comment by GhostofGoes** [comment]
Fixed in #71
